ta_services/secure_key_services/include/sks_ta.h

```diff
@@ -14,13 +14,615 @@
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ * param#1: in-memref : [input-data]
+ * param#2: none
+ * param#3: none
+ *
+ * These commands relate to the PKCS#11 API functions C_SignUpdate() and
+ * C_VerifyUpdate.
+ */
+#define SKS_CMD_SIGN_UPDATE		0x00000019
+#define SKS_CMD_VERIFY_UPDATE		0x0000001a
+
+/*
+ * SKS_CMD_SIGN_FINAL - Initialize a signature computation processing
+ * SKS_CMD_VERIFY_FINAL - Initialize a signature verification processing
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ * param#1: none
+ * param#2: out-memref : [output-data]
+ * param#3: none
+ *
+ * These commands relate to the PKCS#11 API functions C_SignFinal() and
+ * C_SignFinal.
+ */
+#define SKS_CMD_SIGN_FINAL		0x0000001b
+#define SKS_CMD_VERIFY_FINAL		0x0000001c
+
+/*
+ * SKS_CMD_FIND_OBJECTS_INIT - Initialize a objects search
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ *			   [struct sks_object_head attribs + attributes data]
+ * param#1: none
+ * param#2: none
+ * param#3: none
+ *
+ * This command relates to the PKCS#11 API function C_FindOjectsInit().
+ */
+#define SKS_CMD_FIND_OBJECTS_INIT	0x0000001d
+
+/*
+ * SKS_CMD_FIND_OBJECTS - Get handles of matching objects
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ * param#1: none
+ * param#2: out-memref : [uint32_t object_handle[max_handles_number]]
+ * param#3: none
+ *
+ * This command relates to the PKCS#11 API function C_FindOjects().
+ */
+#define SKS_CMD_FIND_OBJECTS		0x0000001e
+
+/*
+ * SKS_CMD_FIND_OBJECTS_FINAL - Finalize current objects search
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ * param#1: none
+ * param#2: none
+ * param#3: none
+ *
+ * This command relates to the PKCS#11 API function C_FindOjectsFinal().
+ */
+#define SKS_CMD_FIND_OBJECTS_FINAL	0x0000001f
+
+/*
+ * SKS_CMD_GET_OBJECT_SIZE - Get size used by object in the TEE
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ *			   [uint32_t key_handle]
+ * param#1: none
+ * param#2: out-memref : [uint32_t object_byte_size]
+ * param#3: none
+ */
+#define SKS_CMD_GET_OBJECT_SIZE		0x00000020
+
+/*
+ * SKS_CMD_GET_ATTRIBUTE_VALUE - Get the value of object attrbiute(s)
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ *			   [uint32_t key_handle]
+ *			   [struct sks_object_head attribs + attributes data]
+ * param#1: none
+ * param#2: out-memref : [struct sks_object_head attribs + attributes data]
+ * param#3: none
+ */
+#define SKS_CMD_GET_ATTRIBUTE_VALUE	0x00000021
+
+/*
+ * SKS_CMD_SET_ATTRIBUTE_VALUE - Set the value for object attrbiute(s)
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ *			   [uint32_t key_handle]
+ *			   [struct sks_object_head attribs + attributes data]
+ * param#1: none
+ * param#2: out-memref : [struct sks_object_head attribs + attributes data]
+ * param#3: none
+ */
+#define SKS_CMD_SET_ATTRIBUTE_VALUE	0x00000022
+
+/*
+ * SKS_CMD_DERIVE_KEY - Create a key by derivation of a provisionned parent key
```

Comment:
s/provisionned/provisioned
or maybe "Derive a key from already provisioned parent key"

Reply:
thank for the proposal